astro/src/content/quickstarts/quickstart-javascript-express-web.mdx

```diff
@@ -0,0 +1,259 @@
+- `state`: A random string created in your homepage route and stored in the cookie as `stateValue`
+- `code_challenge`: Created in the homepage route using `pkce-challenge` and stored in a cookie as `challenge`, this is one half of the PKCE 
+
+### OAuth Redirect
+
+When FusionAuth redirects back to your Express application, it will return to your `/oauth-redirect` route. In this response it will have two query parameters that you need to further examine: `code` and `state`. 
+
+You'll need to do the following:
+
+1. Check that the `stateValue` stored in the cookie matches the `state` query parameter.
+1. Make a request to the FusionAuth token endpoint to exchange the authorization `code` received after the user authenticates at FusionAuth and a `verifier` from our `userSession` cookie for an access token.
+1. Retrieve the user by passing the access token to FusionAuth.
+1. Store access token details in the `userToken` cookie.
+1. Store retrieved user in `userDetails` cookie.
+1. Redirect to `/account` route.
+
+<RemoteCode url={frontmatter.coderoot + "/complete-application/src/index.ts"}
+  lang="ts"
+  tags="oauth-redirect"/>
+
+### Account Route
+
+The `/account` page represents what a user sees when they’re logged into their ChangeBank account. A user that isn’t logged in who tries to access this page should just be taken to the home page.
+
+<Aside type="note">
+In a production application, you would validate the token signature, make sure it wasn't expired, and attempt to refresh it if it were expired.
+</Aside>
+
+<RemoteCode url={frontmatter.coderoot + "/complete-application/src/index.ts"}
+  lang="ts"
+  tags="account"/>
+
+### Make Change
+
+Two functions are needed to accept both GET and POST requests on the `/make-change` route. The `GET` verifies that the user is logged in, then servers the HTML file required for the page. The `POST` takes the amount form value as a body parameter and uses the it to calculate how many quarters, dimes, nickels and pennies to return. If `total` is not a valid dollar value or cannot be converted an `error` message is returned. The return string `message` is received by asyncronously and updates the `change-message` or `error-message` div accordingly.
+
+Add the code below to `index.ts`.
+
+<RemoteCode url={frontmatter.coderoot + "/complete-application/src/index.ts"}
+  lang="ts"
+  tags="make-change"/>
+
+### Logout Route
+
+<Aside type="tip">
+Click the `Logout` button and watch the browser first go to FusionAuth to log out the user, then return to your home page.
+</Aside>
+
+The last step is to implement logout. When you log a user out of an application, you'll take them to 
+FusionAuth's `/oauth2/logout` endpoint. FusionAuth calls the logout endpoint (killing the express session), then kills the FusionAuth session, then redirects to the logout endpoint. This endpoint deletes any cookies that your application created, clearing the user's session.
```

Comment:
I don't understand this. Why do we call the logout endpoint two times? Is it because the first time kills the session and the second time kills cookies? I don't get it.

Reply:
@mooreds right now it is sharing where FusionAuth redirects back with the application calling the same. Would it be cleaner to add an additional endpoint called `/oauth2/logout` so there is less confusion? 

Or I could even change the button to call `${fusionAuthURL}/oauth2/logout?client_id=${clientId}` directly instead, I just didn't want to put any logic into the html.

Reply:
I went for the former and updated.